### ISSUE TITLE
Doubt in implementation.

### ISSUE THREAD

**Issue by BismeetSingh** [aresio/fst-pso#14]
I have a dataset with 303 rows,14 columns, and a target variable. How do I pass this to the function? I think it has something to do with the search space.

**Comment by aresio** [comment]
Hi BismeetSingh,

there are several options for passing this kind of data to the fitness function. First of all, can you please tell me what kind of parameters you are optimizing with FST-PSO?

(If you want to discuss this topic privately, feel free to send me an email to m.s.nobile@tue.nl)

**Comment by BismeetSingh** [comment]
I am trying to do something similar to https://pyswarms.readthedocs.io/en/latest/examples/usecases/train_neural_network.html
Essentially, trying to train a neural network but with fst-pso instead of the pso implementation provided by pyswarms.

**Comment by aresio** [comment]
Hi BismeetSingh,

I understand what you are implementing. As I anticipated to you, there are multiple ways in FST-PSO to provide external data to the fitness function. The simplest/cleanest is to exploit the `arguments` optional parameter of the `set_fitness()` method. This parameter expects a dictionary with the optional information that you need in your fitness function. For instance, you can do:

```
data = loadtxt("mydataset.csv")
FP = FuzzyPSO()
FP.set_fitness(fitness_function, arguments={'dataset': data})
...
```

In the next days I will try to provide you with a concrete example on a neural network.

**Comment by aresio** [comment]
Hi BismeetSingh,

I re-implemented the pyswams' example using FST-PSO. [Please check it out in the repository](https://github.com/aresio/fst-pso/blob/master/examples/training_NN.py). As you can see, the code is basically the same and I pass the data to the fitness evaluation using `arguments`.

The main difference wrt to pyswarms is that they do not explicitly initialize the search space. I used [-5,5] for all dimensions, in my code, and it works very well (I got 99.5% of accuracy in my tests). 

I hope this helps your tests. Let me know if you need additional information.

**Comment by BismeetSingh** [comment]
Thanks, I will check it out.

**Comment by BismeetSingh** [comment]
I am stuck on the keepdims = True argument, as it no longer works, tried implementing a different softmax implementation,but gets the wrong result.

**Comment by aresio** [comment]
https://docs.scipy.org/doc/numpy/reference/generated/numpy.sum.html

According to numpy's documentation, keepdims should still work. I run the code succesfully. Which version of numpy are you using?

**Comment by BismeetSingh** [comment]
https://stackoverflow.com/questions/46420007/numpy-sum-got-an-keepdims-error
This is the exact error I get.

**Comment by aresio** [comment]
Hi, did you solve the issue eventually?

**Comment by BismeetSingh** [comment]
I did but it was a long time ago and I don't remember how.

**Comment by aresio** [comment]
That's okay, at least I can close this issue :) thx
